cpp/src/arrow/flight/internal.cc

```diff
@@ -123,6 +176,25 @@ Status FromGrpcStatus(const grpc::Status& grpc_status) {
   }
 }
 
+Status FromGrpcStatus(const grpc::Status& grpc_status, grpc::ClientContext* ctx) {
+  if (grpc_status.ok()) {
+    return Status::OK();
+  }
+
+  const Status& status = FromGrpcCode(grpc_status);
```

Comment:
I'm not sure `const Status&` is correct, since `FromGrpcCode` returns a copy. You must keep the returned instance alive as a value.